src/Compilers/CSharp/Portable/CSharpExtensions.cs

```diff
@@ -317,6 +317,22 @@ internal static Syntax.InternalSyntax.DirectiveStack ApplyDirectives(this Syntax
             return builder.ToList();
         }
 
+        #region ITypeSymbol
+
+        /// <summary>
+        /// Returns true iff the supplied type is sbyte, byte, short, ushort, int, uint,
+        /// long, ulong, bool, char, string, or an enum-type, or if it is the nullable type
+        /// corresponding to one of those types. These types were permitted as the governing
+        /// type of a switch statement in C# 6.
+        /// </summary>
+        public static bool IsValidV6SwitchGoverningType(this ITypeSymbol type)
```

Comment:
It doesn't "need" to be in any particular place, but this is a language question implemented in the compiler layer. Generally speaking, when language questions need to be asked by other layers, they should do so via public APIs. I think you're suggesting that perhaps this language question is not generally useful enough to expose publicly?

---
In reply to: [77065023](https://github.com/dotnet/roslyn/pull/13498#discussion_r77065023) [](ancestors = 77065023)